tidb-server/main.go

```diff
@@ -161,6 +163,7 @@ func main() {
 	registerStores()
 	registerMetrics()
 	config.InitializeConfig(*configPath, *configCheck, *configStrict, reloadConfig, overrideConfig)
+	initializeTempDir()
```

Comment:
we only need to initialize the temp storage directory when `oom-use-tmp-storage` is set to true. we should check the value before calling this function.